File: tools/parity/paths.py
"""Single source of truth for the paths tools/parity/export_py.py needs.

export_py.py and export_r.R each used to hardcode their own copies of the
repo-relative package path (`sys.path.insert(0, "python")` /
`pkgload::load_all("R/sleepermetrics", ...)`) and the season-data directory
(`apply_playoffs(ss, "playoffs")` -- a directory that never actually existed;
see the restructure notes in CLAUDE.md). Centralizing the Python side here
means a future rename only needs one edit instead of finding every literal.

There is no R equivalent module (R doesn't have a portable "this file's own
directory" primitive the way `Path(__file__)` is): export_r.R instead reads
these same literal values directly, kept in sync with this file by hand. If
you change PY_PACKAGE_DIR or SEASON_DIR here, update export_r.R's own
`pkgload::load_all(...)` and `sl_apply_playoffs(ss, ...)` calls to match.
"""
from __future__ import annotations

from pathlib import Path

# tools/parity/paths.py -> parent = tools/parity/ -> parent.parent = tools/ ->
# parent.parent.parent = repo root (this file is nested two levels deep).
REPO_ROOT = Path(__file__).resolve().parent.parent.parent

#: repo-root-relative dir name of the Python instance's package tree.
PY_PACKAGE_DIR = "fantasy-football-4-fun"

#: repo-root-relative dir name of the durable season-data tree.
SEASON_DIR = "data/seasons"

File: r-analysis/sleepermetrics/R/paths.R
# Single source of truth for this instance's season-data directory --------

#' Resolve the season-data directory
#'
#' Every caller that needs the durable `data/seasons/` tree (custom playoff
#' bracket configs, `<root_league_id>/<season>_season.json`) used to hardcode
#' its own `"season"` default, relative to the current working directory.
#' That meant a repo restructure (or even just running from a different
#' directory) could silently point at a directory that no longer exists.
#' This function is the one place that default is computed; every other
#' caller in this package should use it instead of hardcoding a literal.
#'
#' `data/seasons/` holds ONLY league-scoped bracket configs. The Python side
#' additionally has a sibling `data/sources/` tree (the ADP cache, weekly
#' stat caches, nflverse snapshots, default scoring chart) which this R
#' package has no equivalent of -- see `data/sources/README.md`.
#'
#' Resolution order: the `SLEEPERMETRICS_SEASON_DIR` environment variable if
#' set (mirrors the Python side's own override, so both instances can be
#' pointed at the same directory); otherwise the literal `"data/seasons"`,
#' relative to the current working directory -- which is the repo root when
#' run the standard way (`launch.py --r ...`, `Rscript` from repo root, or
#' `verify.py`'s `testthat::test_local()` call), since `data/seasons/` is a
#' sibling of `r-analysis/` at the repo root, not nested inside it.
#'
#' Opening `r-analysis/FantasyFootball.Rproj` directly in RStudio instead
#' sets the working directory to `r-analysis/`, one level below repo root;
#' in that case pass an explicit path (e.g. `sl_dashboard(playoffs =
#' "../data/seasons")`) or `setwd()` to the repo root first.
#'
#' @return A (possibly non-existent) path string. Callers that need it to
#'   exist should check `dir.exists()` themselves, same as before.
#' @export
sl_season_dir <- function() {
  env <- Sys.getenv("SLEEPERMETRICS_SEASON_DIR", "")
  if (nzchar(env)) return(env)
  "data/seasons"
}
